### PR TITLE
[Graphbolt] Add integration test for sampling in CI.

### PR DESCRIPTION
**Pull request by yxy235**
## Description

Add the test to verify whether the sampling result is as expected.
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tests/python/pytorch/graphbolt/test_sampling.py
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/test_sampling.py:1]: rename to test_integration.py

File: tests/python/pytorch/graphbolt/test_sampling.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
+import dgl
+import dgl.graphbolt as gb
+import dgl.sparse as dglsp
+import torch
+
+
+def test_sampling():
+    def setup_seed(seed):
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/test_sampling.py:8]: Move this util function to https://github.com/dmlc/dgl/blob/master/tests/python/pytorch/graphbolt/gb_test_utils.py

File: tests/python/pytorch/graphbolt/test_sampling.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
+import dgl
+import dgl.graphbolt as gb
+import dgl.sparse as dglsp
+import torch
+
+
+def test_sampling():
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/test_sampling.py:7]: test_integration_link_prediction

Comment (frozenbugs) [reply tests/python/pytorch/graphbolt/test_sampling.py:7]: Maybe also add test_integration_node_classification

File: tests/python/pytorch/graphbolt/gb_test_utils.py [hunk 3]
```diff
@@ -165,3 +166,11 @@ def random_homo_graphbolt_graph(
                     path: {test_path}
     """
     return yaml_content
+
+
+def setup_seed(seed):
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+    torch.backends.cudnn.deterministic = True
+    dgl.seed(seed)
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/gb_test_utils.py:175]: Likely we don't need set seed for dgl and dgl random, can you try remove and see whether it still passes?

**Review by frozenbugs — APPROVED** [review]

File: tests/python/pytorch/graphbolt/test_integration.py [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+            [0.4088, 0.8200, 0.1851],
+            [0.0056, 0.9469, 0.4432],
+            [0.8972, 0.7511, 0.3617],
+            [0.5773, 0.2199, 0.3366],
+        ]
+    )
+
+    item_set = gb.ItemSet(node_pairs, names="node_pairs")
+    graph = gb.from_csc(indptr, indices)
+    # print(graph)
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/test_integration.py:42]: nit, remove debug related code.

File: tests/python/pytorch/graphbolt/test_integration.py [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/test_integration.py:196]: ditto

File: tests/python/pytorch/graphbolt/gb_test_utils.py (hunk 3 above)
Comment (yxy235) [reply tests/python/pytorch/graphbolt/gb_test_utils.py:175]: Right.
